webpack.config.js

```diff
@@ -0,0 +1,349 @@
+/* Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This is the main configuration file for webpack.
+// See: https://webpack.js.org/configuration/
+
+const path = require("path");
+const zlib = require("zlib");
+const glob = require("glob");
+const rtlcss = require("rtlcss");
+const CompressionPlugin = require("compression-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const CssMinimizerPlugin = require("css-minimizer-webpack-plugin");
+const ImageMinimizerPlugin = require("image-minimizer-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const ProvidePlugin = require("webpack").ProvidePlugin;
+const RemoveEmptyScriptsPlugin = require("webpack-remove-empty-scripts");
+const { WebpackManifestPlugin } = require("webpack-manifest-plugin");
+
+/* Shared Plugins */
+
+const sharedCompressionPlugins = [
+  new CompressionPlugin({
+    filename: "[path][base].gz",
+    algorithm: "gzip",
+    compressionOptions: { level: 9, memLevel: 9 },
+    // Only compress files that will actually be smaller when compressed.
+    minRatio: 1,
+  }),
+  /* TODO: Add plugins to compress brotli for text/font files vs the generic
+           Use BROTLI_MODE_TEXT/BROTLI_MODE_FONT and add a `test` qualifier. */
```

Comment:
I tinkered with this a bit locally and didn't find material differences - largely due to Brotli being generally great, and our files are relatively small, so the impact of change is in tens-of-bytes. I opted to keep the configuration simpler as a result.